library/ecdh.c

```diff
@@ -31,6 +31,7 @@
 #include "mbedtls/ecdh.h"
 #include "mbedtls/platform_util.h"
 #include "mbedtls/error.h"
+#include "ssl_misc.h"
```

Comment:
I'm sorry, I don't think that's right. Modules from libmbedcrypto should not `#include` header files that don't belong to the crypto library - we don't want that kind of reverse dependency. libmbedtls depends on libmedcrypto, not the other way round.

(At some point, we may split the crypto library into its own separate project, and at that point we can't have reverse dependencies. Even if if never happens, I think preparing for it just results in a cleaner design.)

I suggest creating a new header file for ECDH functions that are currently declared in `ssl_misc.h`, for example `library/ecdh_ext.h`.

Reply:
Add `ecdh_misc.h`